torch/csrc/python_serialization.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include <Python.h>
+
+// Call Python fildes.read(nbytes) and copy it to buf.
+// Requires a buffer for the bytes object returned from read
+// so the reading is chunked.
+inline ssize_t pyReadBuffered(PyObject* fildes, void* buf, size_t nbytes) {
+  const size_t buffer_size = 262144;  // 2^18
+  size_t read_bytes = 0;
+
+  while (read_bytes < nbytes) {
+    auto remaining = nbytes - read_bytes;
+    auto to_read = remaining > buffer_size ? buffer_size : remaining;
+    THPObjectPtr r(PyObject_CallMethod(fildes, "read", "i", to_read));
+    if (!r) throw python_error();
+
+    // read output is String (Python 2) / Bytes (Python 3)
+#if PY_MAJOR_VERSION >= 3
+    auto size = PyBytes_GET_SIZE(r.get());
+    const void* bytes = PyBytes_AsString(r.get());
+#else
+    auto size = PyString_GET_SIZE(r.get());
+    const void* bytes = PyString_AsString(r.get());
+#endif
+
+    // we read EOF
+    if (size == 0) {
+      return read_bytes;
+    }
+
+    memcpy(reinterpret_cast<char*>(buf) + read_bytes, bytes, size);
+    read_bytes += size;
+  } // Reading loop
+
+  return read_bytes;
+}
+
+static inline bool isUnsupportedOperation() {
+  THPObjectPtr io(PyImport_ImportModule("io"));
+  if (!io) throw python_error();
+  THPObjectPtr exception(PyObject_GetAttrString(io, "UnsupportedOperation"));
+  if (!exception) python_error();
+  return PyErr_ExceptionMatches(exception.get());
+}
+
+// Either does fildes.readinto(buf) or fildes.write(buf)
+static inline ssize_t io(PyObject* fildes, void* buf, size_t nbytes, bool is_read) {
```

Comment:
it's a bit confusing to call this function `io` because it's used as a different kind of symbol elsewhere. How about `doPythonIO`?